pkg/api/defaults-kubelet.go

```diff
@@ -191,6 +194,10 @@ func (cs *ContainerService) setKubeletConfig(isUpgrade bool) {
 			// if upgrade, force default "--pod-infra-container-image" value
 			cs.Properties.MasterProfile.KubernetesConfig.KubeletConfig["--pod-infra-container-image"] = o.KubernetesConfig.KubeletConfig["--pod-infra-container-image"]
 		}
+		//Ensure cloud-provider setting
```

Comment:
We don't need this, this is already being taken care of in L152-L154

Reply:
This is to ensure the KubeletConfig in MasterProfile get updated when we upgrade from a in-tree cloud-provider cluster to a external cloud-provider. Without this line, the previous value for this field will not get overwritten.

Reply:
O.K., thanks for clarifying